### PR TITLE
nxos: Removing local-nxapi plugin and related parameters' usage from the integration tests.

### PR DESCRIPTION
**Pull request by nkshrishail**
Depends-On: https://github.com/ansible/ansible/pull/63014

##### SUMMARY
Removing local-nxapi plugin support, and removing all usages of the same from nxos_* integration tests.

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

**Comment by nkshrishail** [comment]
@mikewiebe @chrisvanheuveln

**Comment by mikewiebe** [comment]
@nkshrishail Looks like there are some minor shippable errors to resolve.

**Comment by nkshrishail** [comment]
@mikewiebe , was under the impression that the error was in shippable itself like how we saw on some of my earlier PRs. More close analysis lead to some yaml lint issues. Should pass this time.

**Comment by nkshrishail** [comment]
@mikewiebe looks good now...

**Comment by chrisvanheuveln** [comment]
shipit

**Comment by nkshrishail** [comment]
@chrisvanheuveln @mikewiebe I will update the subject line of this PR to reflect that we are just removing usage of local-nxapi and related parameters from the integration tests.

**Comment by mikewiebe** [comment]
shipit

File: test/integration/targets/nxos_vxlan_vtep_vni/tests/common/sanity.yaml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ---
 - debug: msg="START connection={{ ansible_connection }} nxos_vxlan_vtep_vni sanity test"
-- debug: msg="Using provider={{ connection.transport }}"
   when: ansible_connection == "local"
```
Comment (mikewiebe) [inline test/integration/targets/nxos_vxlan_vtep_vni/tests/common/sanity.yaml]: I added the `shipit` tag but I missed this.  This `when: ansible_connection == "local"` needs to be removed here and any other test file where it appears.

File: test/integration/targets/nxos_vxlan_vtep/tests/common/sanity.yaml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 ---
 - debug: msg="START connection={{ ansible_connection }} nxos_vxlan_vtep sanity test"
-- debug: msg="Using provider={{ connection.transport }}"
   when: ansible_connection == "local"
```
Comment (mikewiebe) [inline test/integration/targets/nxos_vxlan_vtep/tests/common/sanity.yaml]: Remove `when: ansible_connection == "local"`

File: test/integration/targets/nxos_vtp_version/tests/common/sanity.yaml [hunk 3]
```diff
@@ -1,6 +1,5 @@
 ---
 - debug: msg="START connection={{ ansible_connection }} nxos_vtp_version sanity test"
-- debug: msg="Using provider={{ connection.transport }}"
   when: ansible_connection == "local"
```
Comment (mikewiebe) [inline test/integration/targets/nxos_vtp_version/tests/common/sanity.yaml]: Remove `when: ansible_connection == "local"`

File: test/integration/targets/nxos_vtp_password/tests/common/sanity.yaml [hunk 4]
```diff
@@ -1,6 +1,5 @@
 ---
 - debug: msg="START connection={{ ansible_connection }} nxos_vtp_password sanity test"
-- debug: msg="Using provider={{ connection.transport }}"
   when: ansible_connection == "local"
```
Comment (mikewiebe) [inline test/integration/targets/nxos_vtp_password/tests/common/sanity.yaml]: Remove `when: ansible_connection == "local"`

File: test/integration/targets/nxos_smoke/tests/common/misc_tests.yaml [hunk 5]
```diff
@@ -27,4 +25,3 @@
   nxos_command:
     commands:
       - show hostname
-    provider: "{{ connection | combine({'timeout': 500}) }}"
```
Comment (mikewiebe) [inline test/integration/targets/nxos_smoke/tests/common/misc_tests.yaml]: This one concerns me a bit.  Pretty long timeout here which probably indicates there was a problem on some platforms..  We probably need to use the new method of setting the per task timeout using the `vars:` setting.  One of the 3 options I asked you to test for `nxos_gir`

Check the history for this file to see if the PR that added this indicates why it was added.

**Comment by nkshrishail** [comment]
@NilashishC This is the PR thats cleaning up usage of local-nxapi plugin, once this gets merged, other two which were around removal of "provider" can also go in.

File: test/integration/targets/nxos_nxapi/tests/cli/configure.yaml [hunk 6]
```diff
@@ -22,13 +22,13 @@
         - show nxapi | json
     register: result
 
-  - include: targets/nxos_nxapi/tasks/platform/n7k/assert_changes_https.yaml
+  - include: tasks/platform/n7k/assert_changes_https.yaml
```
Comment (mikewiebe) [inline test/integration/targets/nxos_nxapi/tests/cli/configure.yaml:25]: Thinking about this more and I believe its best not make any changes to tests under `targets/nxos_nxapi` or `targets/nxos_nxapi/transports`.  We should still have some coverage for nxapi local until the code is removed from  `module_utils/nxos.py`.  These two test suites are specific to nxapi validation so these make sense to keep around.

**Review by mikewiebe — CHANGES_REQUESTED** [review]
I added the `shipit` tag a bit early.  After closer review have a few comments.

File: test/integration/targets/nxos_vxlan_vtep_vni/tests/common/sanity.yaml (hunk 1 above)
Comment (nkshrishail) [reply test/integration/targets/nxos_vxlan_vtep_vni/tests/common/sanity.yaml]: Thats my bad, should have paid more attention to all usages of local. Will review again and clean it up. Thanks.

File: test/integration/targets/nxos_smoke/tests/common/misc_tests.yaml (hunk 5 above)
Comment (nkshrishail) [reply test/integration/targets/nxos_smoke/tests/common/misc_tests.yaml]: This was not deliberate, I failed to notice this change in the sea of changes. Thanks for catching it.

**Comment by nkshrishail** [comment]
@mikewiebe I have made further changes, should be good to go.

**Comment by mikewiebe** [comment]
shipit

**Review by mikewiebe — APPROVED** [review]

**Comment by nkshrishail** [comment]
@NilashishC Rebased.

File: lib/ansible/plugins/action/nxos.py [hunk 7]
```diff
@@ -41,8 +41,7 @@ def run(self, tmp=None, task_vars=None):
         self._config_module = True if module_name == 'nxos_config' else False
         socket_path = None
 
-        if (self._play_context.connection == 'httpapi' or self._task.args.get('provider', {}).get('transport') == 'nxapi') \
-                and module_name in ('nxos_file_copy', 'nxos_nxapi'):
+        if self._play_context.connection in ('httpapi', 'local') and module_name in ('nxos_file_copy', 'nxos_nxapi'):
```
Comment (NilashishC) [inline lib/ansible/plugins/action/nxos.py]: [This](https://github.com/ansible/ansible/pull/65876) PR deprecates `provider`. Hence, we would still have to honor provider.transport until it is removed.

File: test/integration/targets/nxos_smoke/tasks/nxapi.yaml [hunk 8]
```diff
@@ -25,9 +25,3 @@
   with_items: "{{ test_items }}"
   loop_control:
     loop_var: test_case_to_run
-
-- name: run test cases (connection=local)
-  include: "{{ test_case_to_run }} ansible_connection=local connection={{ nxapi }}"
```
Comment (NilashishC) [inline test/integration/targets/nxos_smoke/tasks/nxapi.yaml]: We would still want to keep running nxapi with local for nxos_smoke.

**Review by NilashishC — CHANGES_REQUESTED** [review]

**Comment by nkshrishail** [comment]
@NilashishC - Updated the PR per your comments, please take a look.

**Comment by NilashishC** [comment]
bot_status

File: test/integration/targets/nxos_config/tests/common/save.yaml [hunk 9]
```diff
@@ -12,12 +12,11 @@
     parents:
       - "interface {{ intname }}"
     match: none
-    provider: "{{ connection }}"
 
 - name: save config
   nxos_config:
     save_when: always
-    provider: "{{ connection }}"
+    timeout: 300
```
Comment (NilashishC) [inline test/integration/targets/nxos_config/tests/common/save.yaml]: The top-level provider spec has been removed. This is no longer a valid attribute. CI failure:
https://object-storage-ca-ymq-1.vexxhost.net/v1/a0b4156a37f9453eb4ec7db5422272df/ansible_68/65068/f7c47b0036d8aac0f5ca98154b87e1582bb903fd/third-party-check/ansible-test-network-integration-nxos-cli-python36/253142f/controller/ara-report/result/7ab707d4-5474-402e-9c99-e3c5a6b78627/

File: test/integration/targets/nxos_rollback/tests/common/sanity.yaml [hunk 10]
```diff
@@ -1,26 +1,23 @@
...
       - delete backup.cfg
     match: none
-    provider: "{{ connection }}"
   ignore_errors: yes
 
 - name: Create checkpoint file
   nxos_rollback:
     checkpoint_file: backup.cfg
-    provider: "{{ connection }}"
+    timeout: 300
```
Comment (NilashishC) [inline test/integration/targets/nxos_rollback/tests/common/sanity.yaml]: Same as nxos_config. CI failure:
https://object-storage-ca-ymq-1.vexxhost.net/v1/a0b4156a37f9453eb4ec7db5422272df/ansible_68/65068/f7c47b0036d8aac0f5ca98154b87e1582bb903fd/third-party-check/ansible-test-network-integration-nxos-cli-python36/253142f/controller/ara-report/result/eb50e094-3d1c-48ea-9fa8-edbd85873906/

**Comment by NilashishC** [comment]
recheck

**Review by NilashishC — CHANGES_REQUESTED** [review]
Removing `provider` from nxos_smoke and running local-nxapi causes the connection to default to `network_cli`. I believe those need to be added back.

**Comment by NilashishC** [comment]
recheck

**Comment by nkshrishail** [comment]
bot_status

**Review by NilashishC — APPROVED** [review]

**Comment by NilashishC** [comment]
bot_status

**Comment by NilashishC** [comment]
bot_status

**Pull request merged**